src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,69 @@
+package seedu.addressbook.commands;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Edits a person in the address book.
+ */
+public class EditCommand extends Command {
+  public static final String COMMAND_WORD = "edit";
+
+  public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an existing person in the address book. "
+          + "Contact details can be marked private by prepending 'p' to the prefix.\n"
+          + "Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+          + "Example: " + COMMAND_WORD
+          + " John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+  public static final String MESSAGE_SUCCESS = "Person edited: %1$s";
+
+  private final Person toEdit;
+
+  /**
+   * Convenience constructor using raw values.
+   *
+   * @throws IllegalValueException if any of the raw values are invalid
+   */
+  public EditCommand(String name,
+                    String phone, boolean isPhonePrivate,
```

Comment:
Indentation for wrapped lines should be 8 spaces  <== from coding standard provided